string name
---
bool success
string extra_info